src/ThriveWorkerUnit.sol

```diff
@@ -0,0 +1,192 @@
+
+    modifier onlyValidator() {
+        require(validators.contains(msg.sender), "ThriveProtocol: only a validator can perform this action");
+        _;
+    }
+
+    modifier onceReady() {
+        require(ready, "ThriveProtocol: contract is not ready");
+        _;
+    }
+
+    constructor(
+        address _moderator,
+        address _rewardToken,
+        uint256 _rewardAmount,
+        uint256 _maxRewards,
+        uint256 _deadline,
+        uint256 _maxCompletionsPerUser,
+        address[] memory _validators,
+        string memory _validationMetadata,
+        string memory _metadataVersion,
+        string memory _metadata,
+        address _badgeQuery
+    ) {
+        require(_moderator != address(0), "ThriveProtocol: moderator address is required");
+        require(_badgeQuery != address(0), "ThriveProtocol: badgeQuery address is required");
+        require(_deadline > block.timestamp, "ThriveProtocol: deadline must be in the future");
+
+        moderator = _moderator;
+        rewardToken = _rewardToken;
+        rewardAmount = _rewardAmount;
+        maxRewards = _maxRewards;
+        validationRewardAmount = _rewardAmount / 10; // Validator Reward = 10% of contributor reward
+        deadline = _deadline;
+        maxCompletionsPerUser = _maxCompletionsPerUser;
+
+        for (uint256 i = 0; i < _validators.length; i++) {
+            validators.add(_validators[i]);
+        }
+
+        validationMetadata = _validationMetadata;
+        metadataVersion = _metadataVersion;
+        metadata = _metadata;
+        badgeQuery = IBadgeQuery(_badgeQuery);
+    }
+
+    function initialize() external payable onlyModerator {
+        require(!ready, "ThriveProtocol: already initialized");
+        uint256 requiredEther = validators.length() * validationRewardAmount;
+        uint256 totalRequiredEther = maxRewards + requiredEther;
+        require(msg.value >= totalRequiredEther, "ThriveProtocol: insufficient value for validators and contributors");
+        require(IERC20(rewardToken).balanceOf(msg.sender) >= rewardAmount * maxRewards, "ThriveProtocol: insufficient value for contributors");
+
+        IERC20(rewardToken).safeTransferFrom(msg.sender, address(this), rewardAmount * maxRewards);
+        ready = true;
+
+        emit Initialized();
+    }
+
+    function confirm(address contributor, string memory inputValidationMetadata) external onlyValidator onceReady {
+        require(block.timestamp <= deadline, "ThriveProtocol: work unit has expired");
+        require(completions[contributor] < maxCompletionsPerUser, "ThriveProtocol: max completions per user reached");
+        if (assignedAddress != address(0)) {
+            require(contributor == assignedAddress, "ThriveProtocol: contributor is not the assigned address");
+        }
+
+        for (uint256 i = 0; i < requiredBadges.length(); i++) {
+            require(badgeQuery.hasBadge(contributor, requiredBadges.at(i)), "ThriveProtocol: required badge missing");
```

Comment:
this requires all badges, I think we want to have at least one of the badges right?

Reply:
Yes, you are correct. Changed the logic to support if user has at least one of the badges.